heudiconv/dicoms.py

```diff
@@ -193,31 +194,39 @@ def group_dicoms_into_seqinfos(files, file_filter, dcmfilter, grouping):
             sequence_name = dcminfo[0x19, 0x109c].value
         else:
             sequence_name = 'Not found'
+        try:
+            seqinfo_args = (
+                total,
+                op.split(series_files[0])[1],
+                series_id,
+                op.basename(op.dirname(series_files[0])),
+                '-', '-',
+                size[0], size[1], size[2], size[3],
+                TR, TE,
+                dcminfo.ProtocolName,
+                motion_corrected,
+                'derived' in [x.lower() for x in dcminfo.get('ImageType', [])],
+                dcminfo.get('PatientID'),
+                dcminfo.get('StudyDescription'),
+                refphys,
+                dcminfo.get('SeriesDescription'),
+                sequence_name,
+                image_type,
+                accession_number,
+                # For demographics to populate BIDS participants.tsv
+                dcminfo.get('PatientAge'),
+                dcminfo.get('PatientSex'),
+                dcminfo.get('AcquisitionDate'),
+                dcminfo.get('SeriesInstanceUID')
+            )
+            info = SeqInfo(*seqinfo_args)
+        except TypeError as e:
+            print("Trying to construct Seqinfo object:",
+                  "{ob}".format(ob=SeqInfo.__doc__),
+                  "\n However the arguments supplied were:")
+            print([repr(v) for v in seqinfo_args])
```

Comment:
use `lgr.error` not plain print.

Reply:
Yes multiple accession_numbers/studyUIDs being converted as a single study. i.e. the scanner crashes and two "unique sessions" should be merged to prevent needless complication.

Reply:
yep it seems easy enough to always return the uniform object. The only problem arises when you create filegroup_json. If you have duplicate series_ids it will fail which is undesirable for a large number of dicoms passed in that are trivially separable by the combination of series_id and grouping variable (or even just their seqinfo object). We could alter the filegroup json object to have an additionaly layer of nesting, fail with an assertion like i've done, or something more inventive perhaps

Reply:
re: "scanner crashes and two "unique sessions" should be merged"  -- if accession was the same, wouldn't current `--groupping=accession_number` be sufficient? (just trying to get a fuller picture)

Reply:
re: "problem ... filegroup_json. If you have duplicate series_ids "  ""unique sessions" should be merged" ... "undesirable for a large number of dicoms passed in".  Let me ouline
1. I think all DICOMs should be passed in and no magical "merge" (or "substitute") should happen automagically
2. indeed it is a problem (and I think we guard for it) if duplicate `series_id` comes in since `filegroup*.json` contains a dict.  We should make them "non-conflicting" (indeed question how and when?)
3. "by combination of series_id and grouping variable", "additionaly layer of nesting" -- too complex imho and then would not be uniform with already existing setup.  May be we should just disambiguate by adding a "+1", "+2" or alike whenever conflict arises (and issue a warning).  The question is how early such analysis should happen

still digging...  will need to switch to other thing soon though... Question:  do you have (or could you generate e.g. using nifti2dicom) any sample data you could possibly share to demonstrate your usecase?

Reply:
@satra @mgxd -- I just want to make sure: for the usecase when dicom path template provided, should there be additional splitting into sessions/studies within each of those subject(/session) pairs by anything or all matching dicoms should just belong to that subject(/session)?  what would happen if duplicates (canceled sessions, restarted)?

Reply:
re: "wouldn't current --groupping=accession_number be sufficient", the accession_numbers are different too.

re: "(indeed question how and when?)", not sure I know enough about the different ways in which the pipeline can be run. For example an especially challenging condition is if two different tars,  targeted to the same output directory, run at the same time. I guess when filegroup_file is written (save_json(filegroup_file, filegroup)) we could check if it already exists and append to it. This would require a check for duplication and modification of the seqinfo object series_ids too. May be a little bit too challenging. A simpler case to accommodate  is the one I outlined where it is all contained in the same tar.

re: "sample data", I was given this data so I don't know for sure about how it was created. From what I can tell 2 study dicom tars were untarred and the two directories were used to create a new tar. So this process would reasonably replicate the data (provided they have some matching series_ids).